core/modules/ccontrol/testCControl.cc

```diff
@@ -686,6 +707,246 @@ static const std::vector<Antlr4CompareQueries> ANTLR4_COMPARE_QUERIES = {
+              }));
+        },
+        "SELECT objectId FROM Object WHERE (objectID>>10)=1"
+    ),
+
+    // tests the ^ operator
+    Antlr4CompareQueries(
+        "SELECT objectId from Object where objectID ^ 1 = 1",
+        "SELECT objectId from Object where objectID = 1",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+              auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(selectStatement->getWhereClause().getRootTerm());
+              auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+              auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+              auto compPredicate = std::dynamic_pointer_cast<query::CompPredicate>(boolFactor->_terms[0]);
+              compPredicate->left = std::make_shared<query::ValueExpr>(query::ValueExpr::FactorOpVector({
+                  query::ValueExpr::FactorOp(
+                          query::ValueFactor::newColumnRefFactor(query::ColumnRef::newShared("", "", "objectID")),
+                          query::ValueExpr::BIT_XOR),
+                  query::ValueExpr::FactorOp(query::ValueFactor::newConstFactor("1"))
+              }));
+        },
+        "SELECT objectId FROM Object WHERE (objectID^1)=1"
+    ),
+
+    // tests NOT with a BoolFactor
+    Antlr4CompareQueries(
+        "select * from Filter where NOT filterId > 1 AND filterId < 6",
+        "select * from Filter where filterId > 1 AND filterId < 6",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+            // flip the 'not' on the filterId boolFactor to make it 'not'
+            auto whereClauseRef = selectStatement->getWhereClause();
+            auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(whereClauseRef.getRootTerm());
+            auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+            auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+            boolFactor->setHasNot(true);
+        },
+        "SELECT * FROM Filter WHERE NOT filterId>1 AND filterId<6"
+    ),
+
+    // tests NOT with an AND term
+    Antlr4CompareQueries(
+        "select * from Filter where NOT (filterId > 1 AND filterId < 6)",
+        "select * from Filter where (filterId > 1 AND filterId < 6)",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+            // flip the 'not' on the AndTerm to make it 'not'
+            auto whereClauseRef = selectStatement->getWhereClause();
+            auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(whereClauseRef.getRootTerm());
+            auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+            auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+            boolFactor->setHasNot(true);
+        },
+        "SELECT * FROM Filter WHERE NOT(filterId>1 AND filterId<6)"
```

Comment:
This would be easier to read:
```
SELECT * FROM Filter WHERE NOT (filterId > 1 AND filterId < 6)
```

Reply:
this is the expected value from serialization.